includes/wc-cart-functions.php

```diff
@@ -381,3 +460,220 @@ function wc_get_chosen_shipping_method_ids() {
 	}
 	return $method_ids;
 }
+
+/**
+ * Get cart cross sells.
+ * @since 2.7.0
+ * @return array
+ */
+function wc_get_cart_cross_sells() {
+	$cross_sells = array();
+	$in_cart     = array();
+	if ( ! WC()->cart->is_empty() ) {
+		foreach ( WC()->cart->get_cart() as $cart_item_key => $values ) {
+			if ( $values['quantity'] > 0 ) {
+				$cross_sells = array_merge( $values['data']->get_cross_sells(), $cross_sells );
+				$in_cart[] = $values['product_id'];
+			}
+		}
+	}
+	$cross_sells = array_diff( $cross_sells, $in_cart );
+	return $cross_sells;
+}
+
+/**
+ * See if we're displaying tax in the cart in a certain way.
+ * @since 2.7.0
+ * @return boolean
+ */
+function wc_cart_prices_include_tax() {
+	return 'incl' === get_option( 'woocommerce_tax_display_cart', 'excl' );
+}
+
+/**
+ * Determines the value that the customer spent and the subtotal
+ * displayed, used for things like coupon validation.
+ *
+ * Since the coupon lines are displayed based on the TAX DISPLAY value
+ * of cart, this is used to determine the spend.
+ *
+ * If cart totals are shown including tax, use the subtotal.
+ * If cart totals are shown excluding tax, use the subtotal ex tax
+ * (tax is shown after coupons).
+ *
+ * @since 2.7.0
+ * @return string
+ */
+function wc_cart_subtotal_to_display() {
+	return wc_format_decimal( WC()->cart->get_subtotal( wc_cart_prices_include_tax() ) );
+}
+
+/**
+ * Gets the sub total (after calculation).
+ *
+ * @param bool $compound whether to include compound taxes
+ * @return string formatted price
+ */
+function wc_cart_subtotal_html( $compound = false, $echo = true ) {
+	// If the cart has compound tax, we want to show the subtotal as
+	// cart + shipping + non-compound taxes (after discount)
+	if ( $compound ) {
+		$cart_subtotal = WC()->cart->cart_contents_total + WC()->cart->shipping_total + WC()->cart->get_taxes_total( false, false );
+		$suffix = '';
+
+	// Otherwise we show cart items totals only (before discount)
+	} else {
+		$cart_subtotal = wc_cart_subtotal_to_display();
+
+		if ( wc_cart_prices_include_tax() ) {
+			$suffix = wc_cart_prices_include_tax() ? '' : ' <small class="tax_label">' . WC()->countries->ex_tax_or_vat() . '</small>';
```

Comment:
I made changes here and line 531 to put the space before the <small> tag. Having it after the price caused an extra space when there was no suffix.